webpack.config.js

```diff
@@ -0,0 +1,134 @@
+// const BundleAnalyzerPlugin = require('webpack-bundle-analyzer').BundleAnalyzerPlugin
+const fs = require('fs')
+const path = require('path')
+
+const webpackAssets = {
+  javascripts: {
+    chunks: [],
+    bundles: [],
+  },
+}
```

Comment:
Will we record our dependencies like the following pseudo code?
```
const webpackAssets = {
  scrollableImage: {
    chunks: [
      'react-$chunkhash-chunk.js',
      'twreporter-$chunkhash-chunk.js', 
      ... // omit due to explanation only
    ],
    bundles: [
      'scrollable-image-$chunkhash.bundle.js',
    ],
  }
}
```

Otherwise, how do we know what dependencies we need to load before executing?

Reply:
Currently, webpackAssets are consumed in this way:

`
  const packageName = 'scrollable-image'
  const assets = [ ...chunks, ...bundles ]
  const assetScript = assets.map(src => {
      if (src.endsWith('bundle.js')) {
        if (src.indexOf(`${packageName}`) !== -1) {
          return `<script type="text/javascript" src="${src}"></script>`
        }
        return
      }
      return `<script type="text/javascript" src="${src}"></script>`
    })
`

Reply:
The reason of this design is our packages would have common chunks, so the only difference is name of `bundle.js`. Currently, I use the name provided in entry point of webpack config file as the packageName for picking the desired script.

Reply:
Named bundle files are ok for loading.
However, in the above example,
will we load all chunks even if `scrollable-image` does not need it?

Reply:
Since the future repo will contain three packages,
`scrollable-image`, `timeline` and `dual-channel`.

`dual-channel` might generate different chunks which `scrollable-image` doesn't need.
In current `webpack-asset.json`, how to only load necessary chunks for each sub-package?

Reply:
Updated. Please see commit `b959092` and `2715158` for explanation. ( draft #4 is also updated.)

In these patches, the common chunks are in dist/ folder and packages' own chunks would be in their folder (e.g. dist/scrollable-image).

Here's folder structure:

|- dist/
|--dual-channel/
|----dual-channel-chunk.js
|----dual-channel-main.js
|--scrollable-image/
|----scrollable-image-main.js
|--timeline/
|--common-1-chunk.js
|--common-2-chunk.js